src/cpp/configuration/implementations/Address_configuration.cpp

```diff
@@ -30,12 +30,83 @@ Address::Address(
         const RawConfiguration& configuration,
         TransportProtocol default_transport /*= Address::default_transport_protocol()*/)
 {
+    if (!configuration.IsMap() && !configuration.IsNull())
+    {
+        throw ConfigurationException("Address expects a map as base yaml type or an empty one.");
+    }
+
+    // Get IP version. This needs to go first because the DNS call needs to know the IP version
+    if (configuration[ADDRESS_IP_VERSION_TAG])
+    {
+        try
+        {
+            std::string tag = configuration[ADDRESS_IP_VERSION_TAG].as<std::string>();
+
+            if (tag == ADDRESS_IP_VERSION_V4_TAG)
+            {
+                ip_version_ = IPv4;
+            }
+            else if (tag == ADDRESS_IP_VERSION_V6_TAG)
+            {
+                ip_version_ = IPv6;
+            }
+            else
+            {
+                throw ConfigurationException(utils::Formatter() <<
+                              "Error getting Address ip version: it must be <v4> or <v6>");
+            }
+        }
+        catch (const std::exception& e)
+        {
+            throw ConfigurationException(utils::Formatter() <<
+                          "Error getting Address ip version: " << e.what());
+        }
+    }
+    else
+    {
+        // Set ip version default
+        ip_version_ = NONE_IPv;
+    }
+
     // Get IP
     if (configuration[ADDRESS_IP_TAG])
     {
         try
         {
             ip_ = configuration[ADDRESS_IP_TAG].as<IpType>();
+
+            // Check if it is a valid IP or it should call DNS
+            if (!Address::is_ipv4_correct(ip_) && !Address::is_ipv6_correct(ip_))
+            {
+                // Call
+                auto response = fastrtps::rtps::IPLocator::resolveNameDNS(ip_);
+
+                // Add the first valid IP found
+                if (ip_version_ == IPv6)
+                {
+                    if (response.second.size() > 0)
+                    {
+                        ip_ = response.second.begin()->data();
+                    }
+                    else
+                    {
+                        throw ConfigurationException(utils::Formatter() <<
+                          "Incorrect IPv6 set or DNS not found: " << ip_);
+                    }
+                }
+                else
```

Comment:
Here both ``NONE_IPv`` and ``IPv4`` behave the same, that is why I don't think the new category is useful.

Reply:
:+1: